File: src/main/java/me/ste/stevesseries/fancydrops/packet/PacketPlayOutCollectItem.java
package me.ste.stevesseries.fancydrops.packet;

import com.comphenix.protocol.PacketType;
import com.comphenix.protocol.ProtocolLibrary;
import com.comphenix.protocol.events.PacketContainer;

public class PacketPlayOutCollectItem {
    private final int collectedId;
    private final int collectorId;
    private final int pickupItemCount;

    public PacketPlayOutCollectItem(int collectedId, int collectorId, int pickupItemCount) {
        this.collectedId = collectedId;
        this.collectorId = collectorId;
        this.pickupItemCount = pickupItemCount;
    }

    public PacketContainer toContainer() {
        PacketContainer packet = ProtocolLibrary.getProtocolManager().createPacket(PacketType.Play.Server.COLLECT);
        packet.getIntegers().write(0, this.collectedId);
        packet.getIntegers().write(1, this.collectorId);
        packet.getIntegers().write(2, this.pickupItemCount);
        return packet;
    }
}

File: src/main/kotlin/me/ste/stevesseries/fancydrops/packet/PacketPlayOutCollectItem.kt
package me.ste.stevesseries.fancydrops.packet

import com.comphenix.protocol.PacketType
import com.comphenix.protocol.ProtocolLibrary
import com.comphenix.protocol.events.PacketContainer

class PacketPlayOutCollectItem(
    private val collectedId: Int,
    private val collectorId: Int,
    private val pickupItemCount: Int
) : Packet() {
    override val container: PacketContainer
        get() {
            val packet = ProtocolLibrary.getProtocolManager().createPacket(PacketType.Play.Server.COLLECT)

            packet.integers.write(0, this.collectedId)
            packet.integers.write(1, this.collectorId)
            packet.integers.write(2, this.pickupItemCount)

            return packet
        }
}
